src/env/tock/mod.rs

```diff
@@ -41,6 +75,22 @@ impl Env for TockEnv {
     fn user_presence(&mut self) -> &mut Self::UserPresence {
         self
     }
+
+    fn storage(&mut self) -> StorageResult<Self::Storage> {
+        assert_once(&mut self.storage);
+        unsafe { steal_storage() }
+    }
+
+    fn upgrade_storage(&mut self) -> StorageResult<Self::UpgradeStorage> {
+        assert_once(&mut self.upgrade_storage);
+        SyscallUpgradeStorage::new()
```

Comment:
Yes, it's because of the examples. If they use `TockEnv` and I make the PersistentStore return its storage (as the comment you selected), then the asymmetry should disappear.